File: openesdh-repo/src/main/java/dk/openesdh/repo/model/ContactType.scala
package dk.openesdh.repo.model

/**
 * Auto converted by IDEA, Needs to be inspected and (possibly) refactored
 *
 * @author Lanre.
 * @see org.alfresco.service.cmr.security.AuthorityType
 */
object ContactType extends Enumeration {
  type ContactType = Value
  val ORGANIZATION, PERSON, UNSUPPORTED = Value

  def getContactType(contact: String): ContactType = {
    var contactType: ContactType = null

    if (contact == ContactType.ORGANIZATION.toString) {
      contactType = ContactType.ORGANIZATION
    }
    else if (contact == ContactType.PERSON.toString) {
      contactType = ContactType.PERSON
    }
    else contactType = ContactType.UNSUPPORTED

    contactType
  }
}

abstract class ContactType {
  def isFixedString: Boolean

  def getFixedString: String

  def isPrefixed: Boolean

  def getPrefixString: String

  def getOrderPosition: Int

  def equals(contactType: String): Boolean = this equals ContactType.getContactType(contactType)
}

File: openesdh-repo/src/main/java/dk/openesdh/repo/model/ContactType.java
package dk.openesdh.repo.model;

import dk.openesdh.repo.services.contacts.ContactService;
import org.apache.commons.lang.StringUtils;

/**
 * @author lanre.
 */
public enum ContactType {

    ORGANIZATION(true, ContactService.CONTACT_PREFIX, 9),
    PERSON(true, ContactService.CONTACT_PREFIX, 10),
    UNSUPPORTED(false, "", 11);

    private final boolean fixed = false;
    private final String fixedString = "";
    private final boolean prefixed;
    private final String prefixString;
    private final int orderPosition;

    private ContactType(boolean prefixed, String prefixString, int orderPosition) {
        this.prefixed = prefixed;
        this.prefixString = prefixString;
        this.orderPosition = orderPosition;
    }

    public boolean isFixedString() {
        return fixed;
    }

    public String getFixedString() {
        return fixedString;
    }

    public boolean isPrefixed() {
        return prefixed;
    }

    public String getPrefixString() {
        return prefixString;
    }

    public int getOrderPosition() {
        return orderPosition;
    }

    public boolean equals(String authority) {
        return equals(getContactType(authority));
    }

    public static ContactType getContactType(String contact) {
        contact = StringUtils.upperCase(contact);
        if (ContactType.ORGANIZATION.name().equals(contact)) {
            return ContactType.ORGANIZATION;
        }
        if (ContactType.PERSON.name().equals(contact)) {
            return ContactType.PERSON;
        }
        return ContactType.UNSUPPORTED;
    }
}
